File: src/main/java/net/hectus/neobb/turn/default_game/warp/TVoidWarp.java
package net.hectus.neobb.turn.default_game.warp;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.*;
import org.bukkit.World;

import java.util.List;

public class TVoidWarp extends WarpTurn {
    public TVoidWarp(World world) { super(world, "void"); }
    public TVoidWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "void"); }

    @Override
    public int chance() {
        return 10;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(ColdClazz.class, NeutralClazz.class, WaterClazz.class, RedstoneClazz.class, SupernaturalClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/warp/TVoidWarp.kt
package net.hectus.neobb.turn.default_game.warp

import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.turn.default_game.attribute.clazz.*
import kotlin.reflect.KClass

class TVoidWarp(data: PlacedStructure?, player: NeoPlayer?) : WarpTurn(data, "void", player) {
    override val cost: Int = 4
    override val chance: Double = 10.0
    override val allows: List<KClass<out Clazz>> = listOf(ColdClazz::class, NeutralClazz::class, WaterClazz::class, RedstoneClazz::class, SupernaturalClazz::class)
    override val temperature: Temperature = Temperature.COLD
}